include/api/ElasticsearchStateIndex.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#ifndef INCLUDED_ml_api_ElasticsearchStateIndex_h
+#define INCLUDED_ml_api_ElasticsearchStateIndex_h
+
+#include <string>
+
+namespace ml {
+namespace api {
+//! Elasticsearch index for state
+static const std::string ML_STATE_INDEX(".ml-state");
```

Comment:
We should delete this. The c++ should not need to know which index we're persisting into.

Reply:
Ignore this in the light of Dave's comment to do this as a refactoring later.